src/nvim/msgpack_rpc/channel.c

```diff
@@ -834,6 +835,7 @@ static void log_client_msg(uint64_t channel_id,
                            msgpack_object msg)
 {
   DLOGN("[msgpack-rpc] client(%" PRIu64 ") -> nvim ", channel_id);
+  log_lock();
```

Comment:
msgpack logging is only enabled if MIN_LOG_LEVEL >= DEBUG_LOG_LEVEL. But your'e right, we should also consider `DISABLE_LOG`, I will fix the commit